sim/midas/src/main/cc/main.cc

```diff
@@ -73,7 +46,15 @@ int main(int argc, char **argv) {
     // This file can be included in the setup method of any top-level to pass
     // an instance of each driver to the `add_bridge_driver` method. Drivers can
     // be distinguished by overloading the method with the appropriate type.
-    #include "constructor.h"
+    #define GET_CORE_CONSTRUCTOR
```

Comment:
I think a brief comment here calling out that this enforces an instantiation order would be good.